Firestore/Source/Public/FIRFirestore.h

```diff
@@ -115,10 +115,11 @@ NS_SWIFT_NAME(Firestore)
  * and collections. Unlike other firestore access, data accessed with the transaction will not
  * reflect local changes that have not been committed. For this reason, it is required that all
  * reads are performed before any writes. Transactions must be performed while online. Otherwise,
- * reads will fail, and the final commit will fail.
+ * reads will fail, the final commit will fail, and the completion block will return an error.
  *
  * @param updateBlock The block to execute within the transaction context.
- * @param completion The block to call with the result or error of the transaction.
+ * @param completion The block to call with the result or error of the transaction. This
+ *     block will always run eventually.
```

Comment:
@wilhuff unsure if this is something we're formally guaranteeing.

Reply:
Within the scope of the process yes. If the OS restarts us then of course we can't promise that.

You could phrase slightly differently to put less emphasis on the eventuality and instead highlight that unlike other completion handlers for writes this one will run online or off.